src/codegen/program.lisp

```diff
@@ -236,6 +239,11 @@ Example:
                      &optional))
             ,name)))
 
+   ;; Emit inline declamations for functions declared inline at definition-site
+   (loop :for (name . node) :in bindings
```

Comment:
One concern I have is that while we deal with unrolling etc. in the inliner, Lisp may not do so as well. So all the inliner machinery can fail when Lisp itself attempts to inline recursive functions. Probably not something we have to worry about now but it will bite us at some point later. In those cases we should simply not inline at the Lisp level, I think.